### PR TITLE
Improve `rename_like`

### PR DESCRIPTION
**Pull request by malmans2**
`rename_like` also updates the `bounds` and `cell_measures` attributes.

### CODE REVIEW

File: cf_xarray/accessor.py [hunk 1]
```diff
@@ -1332,24 +1332,48 @@ def rename_like(
...
 
-        conflicts = {}
-        for k0, v0 in keydict.items():
-            if len(v0["ours"]) > 1 or len(v0["theirs"]) > 1:
-                conflicts[k0] = v0
-                continue
-            for v1 in keydict.values():
-                # Conflicts have same ours but different theirs or vice versa
-                if (v0["ours"] == v1["ours"]) != (v0["theirs"] == v1["theirs"]):
+        def get_renamer_and_conflicts():
```
Comment (dcherian) [inline cf_xarray/accessor.py:1343]:
```suggestion
        def get_renamer_and_conflicts(keydict):
```

File: cf_xarray/accessor.py [hunk 2]
```diff
@@ -1332,24 +1332,48 @@ def rename_like(
...
+                v["ours"][0]: v["theirs"][0]
+                for k, v in keydict.items()
+                if k not in conflicts
+            }
+
+            return renamer, conflicts
+
+        # Run get_renamer_and_conflicts twice.
+        # The second time add the bounds associated with variables to rename
+        renamer, conflicts = get_renamer_and_conflicts()
```
Comment (dcherian) [inline cf_xarray/accessor.py:1365]:
```suggestion
        renamer, conflicts = get_renamer_and_conflicts(keydict)
```

File: cf_xarray/accessor.py [hunk 3]
```diff
@@ -1332,24 +1332,48 @@ def rename_like(
...
+        if "bounds" not in skip:
+            for k, v in renamer.items():
+                ours = set(getattr(self, "bounds", {}).get(k, []))
+                theirs = set(getattr(other.cf, "bounds", {}).get(v, []))
+                if ours and theirs:
+                    ours.update(keydict.get(k, {}).get("ours", []))
+                    theirs.update(keydict.get(k, {}).get("theirs", []))
+                    keydict[k] = dict(ours=list(ours), theirs=list(theirs))
+            renamer, conflicts = get_renamer_and_conflicts()
```
Comment (dcherian) [inline cf_xarray/accessor.py:1374]:
```suggestion
            renamer, conflicts = get_renamer_and_conflicts(keydict)
```

**Review by dcherian — CHANGES_REQUESTED** [review]

**Review by dcherian — APPROVED** [review]

**Comment by dcherian** [comment]
Docs failure is related to this PR

**Comment by malmans2** [comment]
> Docs failure is related to this PR

I don't think so, main is failing as well. I'm not sure what change (I guess in `xarray`?) but this would fix it: https://github.com/xarray-contrib/cf-xarray/pull/227#discussion_r637399243

**Pull request merged**